newrelic/hooks/mlmodel_langchain.py

```diff
@@ -255,10 +257,138 @@ def wrap_similarity_search(wrapped, instance, args, kwargs):
     return response
 
 
+def wrap_tool_sync_run(wrapped, instance, args, kwargs):
+    transaction = current_transaction()
+    if not transaction:
+        return wrapped(*args, **kwargs)
+
+    # Framework metric also used for entity tagging in the UI
+    transaction.add_ml_model_info("Langchain", LANGCHAIN_VERSION)
+
+    run_args = bind_args(wrapped, args, kwargs)
+    tool_input = run_args.get("tool_input", "")
+    tags = run_args.get("tags", "")
```

Comment:
If we just grab the tags and metadata from the input args we could be missing tags. See https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/chains/base.py#L139. Note it passes self.tags and tags and self.metadata and metadata into the callback manager and those ultimately get combined together and passed to each callback.